includes/discount-functions.php

```diff
@@ -1155,8 +1155,150 @@ function _edd_discount_get_post_doing_it_wrong( $query ) {
+ *
+ * @since 3.0
+ *
+ * @param string $request SQL request.
+ * @param WP_Query $query Instance of WP_Query.
+ *
+ * @return string $request Rewritten SQL query.
+ */
+function _edd_discounts_bc_posts_request( $request, $query ) {
+	if ( 'posts_request' !== current_filter() ) {
+		$message = __( 'This function is not meant to be called directly. It is only here for backwards compatibility purposes.', 'easy-digital-downloads' );
+		_doing_it_wrong( __FUNCTION__, $message, '3.0' );
+	}
+
+	if ( 'edd_discount' === $query->get( 'post_type' ) ) {
+		$defaults = array(
+			'number'  => 30,
+			'status'  => array( 'active', 'inactive', 'expired' ),
+			'order'   => 'DESC',
+			'orderby' => 'date_created',
+		);
+
+		$args = array(
+			'number' => $query->get( 'posts_per_page' ),
+			'status' => $query->get( 'post_status', array( 'active', 'inactive' ) ),
+		);
+
+		$orderby = $query->get( 'orderby', false );
+		if ( $orderby ) {
+			switch ( $orderby ) {
+				case 'none':
+				case 'ID':
+				case 'author':
+				case 'post__in':
+					$args['orderby'] = 'id';
+					break;
+				case 'date':
+					$args['orderby'] = 'date_created';
+					break;
+				default:
+					$args['orderby'] = 'id';
+					break;
+			}
+		}
+
+		$offset = $query->get( 'offset', false );
+		if ( $offset ) {
+			$args['offset'] = $offset;
+		}
+
+		if ( 'any' === $args['status'] ) {
+			$args['status'] = $defaults['status'];
+		}
+
+		$args = wp_parse_args( $args, $defaults );
+
+		$args['offset'] = absint( $args['offset'] );
+		$args['number'] = absint( $args['number'] );
+
+		$table_name = EDD()->discounts->table_name;
+
+		$where = '';
+
+		$request = "SELECT id FROM {$table_name} $where ORDER BY {$args['orderby']} {$args['order']} LIMIT {$args['offset']}, {$args['number']};";
+	}
+
+	return $request;
+}
+add_filter( 'posts_request', '_edd_discounts_bc_posts_request', 10, 2 );
+
+/**
+ * Fill the returned WP_Post objects with the data from the discounts table.
+ *
+ * @since 3.0
+ *
+ * @param array $posts Posts returned from the SQL query.
+ * @param WP_Query $query Instance of WP_Query.
+ *
+ * @return array New WP_Post objects.
+ */
+function _edd_discounts_bc_posts_results( $posts, $query ) {
+	if ( 'posts_results' !== current_filter() ) {
+		$message = __( 'This function is not meant to be called directly. It is only here for backwards compatibility purposes.', 'easy-digital-downloads' );
+		_doing_it_wrong( __FUNCTION__, $message, '3.0' );
+	}
+
+	if ( 'edd_discount' === $query->get( 'post_type' ) ) {
+		$new_posts = array();
+
+		foreach ( $posts as $post ) {
+			$discount = edd_get_discount( $post->id );
+
+			$object_vars = array(
+				'ID'          => $discount->id,
+				'post_title'  => $discount->name,
+				'post_status' => $discount->status,
+				'post_type'   => 'edd_discount',
+				'post_date'   => $discount->date_created,
+			);
+
+			foreach ( $object_vars as $object_var => $value ) {
```

Comment:
`$object_vars` is not defined here.

Reply:
Defined here: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/6224/commits/f93684c65c64c00e2585935bade8c29a244b15eb#diff-45ddc6533456059e09c2dbdec2ec8930R1282 :)

Reply:
Welp, I'm blind :D